finality-aleph/src/network/split.rs

```diff
@@ -159,11 +179,9 @@ impl<
 {
     type S = LeftSender<LeftData, RightData, S>;
     type R = LeftReceiver<LeftData, RightData, R>;
-    fn sender(&self) -> &Self::S {
-        &self.sender
-    }
-    fn receiver(&self) -> Arc<Mutex<Self::R>> {
-        self.receiver.clone()
+
+    fn into(self) -> (Self::S, Self::R) {
```

Comment:
Make `Left/RightNetwork` into instances of `SimpleNetwork` and return `impl ComponentNetwork` to avoid leaking types.